### PR TITLE
Few touch-ups to the docs on web forms in Next

### PR DESCRIPTION
**Pull request by mahmoud-moursy**
## Beyond the minor changes...

- Added a warning that back-end validation is more secure
- Fixed a RegEx that I'm damn-near sure is wrong.
- Returned status codes in the first back-end serverless example
- Added a link the user can click on to learn what a HTTP method is

## Changes that must be reviewed:

- Does the `script` tag need explanation? I think all that needs to be said is that you can indeed use it in Next.JS, without much trouble.

## Documentation / Examples

- [x] Make sure the linting passes by running `yarn lint`

### CODE REVIEW

**Comment by T-O-R-U-S** [comment]
Repo refuses to clone on my end; I'm incapable of running `yarn lint`. Would be grateful if somebody else could do so for me :)

**Comment by T-O-R-U-S** [comment]
Managed to get it sorted; passes the lint check!

File: docs/guides/building-forms.md [hunk 1]
```diff
@@ -180,13 +181,18 @@ export default function handler(req, res) {
...
   if (!body.first || !body.last) {
     return res.json({ data: 'First or last name not found' })
+    // Sends a HTTP bad request error code
+    return res.status(400)
   }
 
   // Found the name.
   res.json({ data: `${body.first} ${body.last}` })
+  // Sends a HTTP success code
+  res.status(200)
```
Comment (ijjk) [inline docs/guides/building-forms.md:195]:
```suggestion
  // Sends a HTTP success code
  res.status(200).json({ data: `${body.first} ${body.last}` })
```

File: docs/guides/building-forms.md [hunk 2]
```diff
@@ -180,13 +181,18 @@ export default function handler(req, res) {
   // in the command line where next.js app is running.
   console.log('body: ', body)
 
-  // Both of these are required.
+  // Guard clause checks for first and last name,
+  // and returns early if they are not found
   if (!body.first || !body.last) {
     return res.json({ data: 'First or last name not found' })
+    // Sends a HTTP bad request error code
+    return res.status(400)
```
Comment (ijjk) [inline docs/guides/building-forms.md:189]:
```suggestion
    // Sends a HTTP bad request error code
    return res.status(400).json({ data: 'First or last name not found' })
```

File: docs/guides/building-forms.md [hunk 3]
```diff
@@ -118,6 +118,7 @@ The following example shows using JavaScript to validate a form:
 </script>
 ```
 
+<!--- Why is this here? If you don't know what the <script> tag is I don't think you would be diving into full-stack with Next... -->
```
Comment (ijjk) [inline docs/guides/building-forms.md:121]:
```suggestion
```

**Review by ijjk — APPROVED** [review]
Thanks for the PR!

**Pull request merged**